packages/block-library/src/search/style.scss

```
@@ -70,7 +70,6 @@ $button-spacing-y: math.div($grid-unit-15, 2); // 6px
 
 	.wp-block-search__input {
 		transition-duration: 300ms;
```

From my understanding... adding an explicit default width here

```suggestion
		transition-duration: 300ms;
		width: 250px;
```

And setting `flex-basis` to `100%` for `.wp-block-search__input` [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/search/style.scss#L93) should help to achieve the desired effect...

---

I don't think setting default width is a good idea considering the fact the this block's width can be set by the user manually. Are you sure that the DOM would handle it in the right & efficient way when user set the width of the search block? incase if we agree to this approach then should there be a fallback when enough room is not available for the block?

---

From my testing, overrides from dashboard still work. Without explicit width the transition don't seem to work.
But yes it can cause issues when sufficient room isn't available (have to check how the current implementation handles this case)

---

If it does not works as expected when enough room is not available then in that case maybe we will have to choose not setting the default width over prioritising the transition, otherwise your suggestion is the best approach that we should proceed with😇